src/components/DuetRoom.tsx

```diff
@@ -142,8 +142,15 @@ const DuetRoom: React.FC<{ maybeRoomId: string | null; isCreate: boolean }> = ({
   const partsSelection = getPartsSelection(roomState);
   const myPart = getMyPart(roomState, playerId);
 
-  // Calculate keyboard dimension
+  // song information
   const tracks = chosenSongMIDI.tracks;
+  const bpm = chosenSongMIDI.header?.tempos[0].bpm;
```

Comment:
Can we move these variables into the if condition where waterfall is used, inside the `middleBox` function? I think it doesn't actually affect the correctness so it should be fine even if you don't though.

Also, what if bpm, beatsPerBar, etc. are missing? Or are they definitely included in every JSON file?

Reply:
Sure! 

The information should be in every JSON file since we need it for the playback. 😅